Move Linux WPT from Buildbot to Taskcluster

SimonSapin opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/22435)
<!-- Reviewable:end -->

SimonSapin committed:
Move linux WPT from Buildbot to Taskcluster

SimonSapin committed:
Taskcluster: align Linux WPT task with what’s now on Buildbot

highfive commented:
Heads up! This PR modifies the following files:
 * @wafflespeanut: python/servo/testing_commands.py

SimonSapin commented:
@bors-servo try=wpt

bors-servo commented:
:hourglass: Trying commit f785a44a5d581626bc874f1c2a4ba9895ef0fad0 with merge 86267e558a0ec2d9613c01d9aba4edd736d9f79d...

SimonSapin commented:
@bors-servo try=wpt

bors-servo commented:
:hourglass: Trying commit 2732c7fd673fe9de963a301402c975f180c6891d with merge 18e852b26ba0cb6f73f65d62593d719dcade0401...

SimonSapin commented:
Hmm, there are still `linux-rel-wpt` and `linux-rel-css` scheduled on buildbot, despite having deployed https://github.com/servo/saltfs/pull/925 and having run `buildbot restart --clean --nodaemon /home/servo/buildbot/master &`

bors-servo commented:
:broken_heart: Test failed - [status-taskcluster](https://tools.taskcluster.net/task-group-inspector/#/cyox3P8aSTeKllOdfiBDew)

SimonSapin commented:
Ah never mind, https://github.com/servo/saltfs/pull/925 did not disable `linux-rel-wpt`. Let’s try this:

@bors-servo try=linux

SimonSapin commented:
@bors-servo try-

SimonSapin commented:
@bors-servo try=linux

SimonSapin commented:
@bors-servo try=linux

bors-servo commented:
:hourglass: Trying commit 8e8b001ad1405471ecdd4dbb4a7057ef0c1c1c94 with merge bba9b9bd5d8f9dad8e526f8d39565f74171a9f4b...

SimonSapin commented:
Good, https://build.servo.org/grid shows nothing for bba9b9b. (Unlike 18e852b, although those were no-op jobs that finished in ~2 seconds.) Trychooser works correctly with Taskcluster!

Back to this PR, `./mach filter-intermittents --reporter-api default` fails because it tries to parse the output of `git log --merges --oneline -1`, but somehow that output is empty. I did not manage to reproduce this outside of Taskcluster, even in a Docker image very similar to what these tasks run.

@jdm, any idea how that output could be empty?

bors-servo commented:
:broken_heart: Test failed - [status-taskcluster](https://tools.taskcluster.net/task-group-inspector/#/TXmEBKKVS42qZf6ynB1mzA)

SimonSapin commented:
Eureka! These tasks run in a shallow clone (`--depth 1`), this is probably what affects `git log --merges --oneline -1`.

https://github.com/servo/servo/blob/a86c23cdcacf8965c6711e644338d644160e99f6/etc/taskcluster/decisionlib.py#L568-L571

SimonSapin commented:
@bors-servo try=wpt

SimonSapin commented:
@bors-servo try=wpt

SimonSapin committed:
Fix `./mach filter_intermittents` in shallow git clones.

`git log --merges` only shows commits with at least two parents.
Presumably, this only works when the parent commits are in the local repository.
In a shallow clone (`--depth 1`), on the HEAD commit is fetched.

This commit removes the `--merges` filter.
This shouldn’t affect the behavior,
since CI always runs with a merge commit as HEAD.

bors-servo commented:
:hourglass: Trying commit 753cf24f902a1ac96630351335a696bbc16274b9 with merge e34b5be80ecf2378c6dd6552847f1d720f136c77...

bors-servo commented:
:broken_heart: Test failed - [status-taskcluster](https://tools.taskcluster.net/task-group-inspector/#/H5P885ftQmejqs___ZWYcg)

SimonSapin commented:
@bors-servo try=wpt

bors-servo commented:
:hourglass: Trying commit e60e534d23209c944b48596d437a9a9c09f20542 with merge b7c666d72730fef873b809c2e2704f2bf5e34f29...

bors-servo commented:
:sunny: Test successful - [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/11038), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/11094), [status-taskcluster](https://tools.taskcluster.net/task-group-inspector/#/dqr2ugQIR2S7SAZ09xsKDA)
State: approved= try=True

SimonSapin commented:
r? @jdm 

I’ve seen intermittent failures on previous runs, but I assume/hope they are no more frequent than on Buildbot. The usual `filtered-wpt-errorsummary.log` is in the Artifacts tab of each WPT task. If something goes really wrong, we can easily revert the `Move linux WPT from Buildbot to Taskcluster` commit until we figure it out.

jdm commented:
@bors-servo r+

bors-servo commented:
:pushpin: Commit e60e534 has been approved by `jdm`

<!-- @bors-servo r=jdm e60e534d23209c944b48596d437a9a9c09f20542 -->

bors-servo commented:
:hourglass: Testing commit e60e534d23209c944b48596d437a9a9c09f20542 with merge ef8be44e46f27ab789de892408617ef11c4f035e...

bors-servo commented:
:sunny: Test successful - [android-mac](http://build.servo.org/builders/android-mac/builds/664), [arm32](http://build.servo.org/builders/arm32/builds/13516), [arm64](http://build.servo.org/builders/arm64/builds/13548), [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/11041), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/11097), [mac-rel-css1](http://build.servo.org/builders/mac-rel-css1/builds/6121), [mac-rel-css2](http://build.servo.org/builders/mac-rel-css2/builds/6099), [mac-rel-wpt1](http://build.servo.org/builders/mac-rel-wpt1/builds/10374), [mac-rel-wpt2](http://build.servo.org/builders/mac-rel-wpt2/builds/10413), [mac-rel-wpt3](http://build.servo.org/builders/mac-rel-wpt3/builds/6176), [mac-rel-wpt4](http://build.servo.org/builders/mac-rel-wpt4/builds/6102), [magicleap](http://build.servo.org/builders/magicleap/builds/274), [status-taskcluster](https://tools.taskcluster.net/task-group-inspector/#/TVwysCccRVe0_CMzvtKZ9A)
Approved by: jdm
Pushing ef8be44e46f27ab789de892408617ef11c4f035e to master...
